x/concentrated-liquidity/incentives_test.go

```diff
@@ -665,6 +698,14 @@ func (s *KeeperTestSuite) TestCalcAccruedIncentivesForAccum() {
 
 				s.Require().Equal(tc.expectedResult, actualResult)
 				s.Require().Equal(tc.expectedIncentiveRecords, updatedPoolRecords)
+
+				// If incentives are fully emitted, we ensure they are cleared from state
+				if tc.recordsCleared {
+					updatedRecordsInState, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, tc.poolId)
+					s.Require().NoError(err)
+
+					s.Require().Equal(0, len(updatedRecordsInState))
+				}
```

Comment:
How is this check useful if `calcAccruedIncentivesForAccum` is non-mutative?

From what I understand, the clearing happens in the `setIncentiveRecord` when called from `updateUptimeAccumulatorsToNow` after calling `calcAccruedIncentivesForAccum`.

I think we should have this checked after testing and calling `updateUptimeAccumulatorsToNow`. WDYT?